### PR TITLE
terraforming 0.15.0 (new formula)

### PR DESCRIPTION
**Pull request by Ketouem**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by apjanke** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

File: Formula/terraforming.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  homepage "http://terraforming.dtan4.net/"
+  url "https://github.com/dtan4/terraforming.git",
+      :tag => "v0.14.0",
+      :revision => "5e934a1a8d18062f9c4e4e3f4a52dd648fc9976d"
+  head "https://github.com/dtan4/terraforming.git"
+
+  def install
+    ENV["GEM_HOME"] = libexec
+    system "gem", "build", "terraforming.gemspec"
+    system "gem", "install", "terraforming-0.14.0.gem"
```
Comment (ilovezfs) [inline Formula/terraforming.rb]: The gem dependencies need to be vendored into the formula. See `braid`, `ghi`, `haste-client`, and `mikutter` for examples.

File: Formula/terraforming.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    bin.install libexec/"bin/terraforming"
+    bin.env_script_all_files(libexec/"bin", :GEM_HOME => ENV["GEM_HOME"])
+  end
+
+  test do
+    system "#{bin}/terraforming", "--help"
```
Comment (ilovezfs) [inline Formula/terraforming.rb]: The test needs to exercise and verify the functionality of the software.

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/terraforming.rb (hunk 2 above)
Comment (Ketouem) [reply Formula/terraforming.rb]: Testing more features of this software would imply connecting to an existing AWS account, except asserting the output of the `version` command I don't really know what kind of test can be performed.

Comment (ilovezfs) [reply Formula/terraforming.rb]: You can also check for an anticipated failure case.

**Comment by Ketouem** [comment]
@ilovezfs The formula now explicitly fetches all dependencies and performs a simple test. Also the installed version is `0.15.0` as a new one has been released since the initial issue date of this PR.

File: Formula/terraforming.rb [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    end
+    system "gem", "build", "terraforming.gemspec"
+    system "gem", "install", "--ignore-dependencies",
+           "terraforming-#{version}.gem"
+    bin.install libexec/"bin/terraforming"
+    bin.env_script_all_files(libexec/"bin", :GEM_HOME => ENV["GEM_HOME"])
+  end
+
+  test do
+    output = shell_output("#{bin}/terraforming help ec2", 0)
```
Comment (ilovezfs) [inline Formula/terraforming.rb]: 0 is already the default

File: Formula/terraforming.rb [hunk 4]
```diff
@@ -0,0 +1,113 @@
+class Terraforming < Formula
+  desc "Export existing AWS resources to Terraform style (tf, tfstate)"
+  homepage "http://terraforming.dtan4.net/"
+  url "https://github.com/dtan4/terraforming.git",
+      :tag => "v0.15.0",
+      :revision => "f4e30400ae61381077b6c78b00b02ddc4642a331"
+  head "https://github.com/dtan4/terraforming.git"
```
Comment (ilovezfs) [inline Formula/terraforming.rb]: I'd remove this since the gem dependencies can fluctuate.

File: Formula/terraforming.rb [hunk 5]
```diff
@@ -0,0 +1,113 @@
+class Terraforming < Formula
+  desc "Export existing AWS resources to Terraform style (tf, tfstate)"
+  homepage "http://terraforming.dtan4.net/"
+  url "https://github.com/dtan4/terraforming.git",
+      :tag => "v0.15.0",
```
Comment (ilovezfs) [inline Formula/terraforming.rb]: can we use the source archive tarball?

File: Formula/terraforming.rb [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    end
+    system "gem", "build", "terraforming.gemspec"
+    system "gem", "install", "--ignore-dependencies",
+           "terraforming-#{version}.gem"
+    bin.install libexec/"bin/terraforming"
+    bin.env_script_all_files(libexec/"bin", :GEM_HOME => ENV["GEM_HOME"])
+  end
+
+  test do
+    output = shell_output("#{bin}/terraforming help ec2")
```
Comment (ilovezfs) [inline Formula/terraforming.rb]: This should do something other than run help. See `faas-cli` for an example of how to mock a connection.

File: Formula/terraforming.rb (hunk 5 above)
Comment (Ketouem) [reply Formula/terraforming.rb]: Unfortunately no, as https://github.com/dtan4/terraforming/blob/master/terraforming.gemspec#L17 implies that the gem building process would take place in a git repo.

Comment (ilovezfs) [reply Formula/terraforming.rb]: ok

File: Formula/terraforming.rb (hunk 6 above)
Comment (Ketouem) [reply Formula/terraforming.rb]: The example that you point to is spawning a TCP server listening on `localhost`, in this formula's case the software will try to reach Amazon's server and there seems to be no way to change the target url.

**Comment by robinbowes** [comment]
What was the blocker here? I'm looking at terraforming right now and `brew install terraforming` would be a real nice-to-have.

**Comment by Ketouem** [comment]
@robinbowes no blocker here, I had to implement test for the formula to get accepted but don't really know how to mock the AWS API.

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Hopefully someone can help write a more robust test here, but let's merge this as-is for now.

**Comment by ilovezfs** [comment]
Thanks for the 🆕 formula @Ketouem! Shipped.

**Pull request closed without merging**